src/extension/webView.ts

```diff
@@ -70,6 +71,27 @@ function validateParameters(options: Options): string[] {
   return validationErrors;
 }
 
+async function getNpmVersion() {
+  return new Promise((resolve, reject) => {
+    exec('npm --version', (error, stdout) => {
+      if (error) {
+        reject(error);
+        return;
+      }
+      const matches = stdout.split('.')[0].match(/\d+/g);
```

Comment:
Version should return a string. It should be something like `6.1.2`.

Also this thing does not work if the output is `v1.2.3`. You must match the version via a regular expression.

Reply:
I am not sure, how we want to compare a string with the number 7. if you don't want to have the version as an integer.

And I am matching the version via a regular expression. it works with v1.2.3 and v10.2.3.

![Screenshot 2022-05-06 095747](https://user-images.githubusercontent.com/55950527/167091656-912d45e4-12ed-4cd5-b5b4-2554e6bf3a7e.png)

Reply:
What you have there does not work. It creates multiple results and you don't know how they are separated.

What you want is `\d+\.\d+\.\d+` - so the one match will give you the full substring that we refer to as "version".

Reply:
> I am not sure, how we want to compare a string with the number 7

By converting it.

1. You get the version (e.g., `"1.2.3"`)
2. You get the major part of (e.g., `"1"`) - this is just a split on the version by "." - first segment
3. You convert to a number via `+` or `*1`.